src/Storages/ObjectStorage/DataLakes/Iceberg/IcebergMetadata.cpp

```diff
@@ -0,0 +1,429 @@
+{
+    auto configuration_ptr = configuration.lock();
+
+    auto snapshots = metadata->get("snapshots").extract<Poco::JSON::Array::Ptr>();
+
+    auto current_snapshot_id = metadata->getValue<Int64>("current-snapshot-id");
+
+    LOG_DEBUG(&Poco::Logger::get("IcebergMetadata initialize"), "Current snapshot id {}", current_snapshot_id);
+
+    for (size_t i = 0; i < snapshots->size(); ++i)
+    {
+        const auto snapshot = snapshots->getObject(static_cast<UInt32>(i));
+        LOG_DEBUG(
+            &Poco::Logger::get("IcebergMetadata initialize"),
+            "Iterationg on snapshot with id {}",
+            snapshot->getValue<Int64>("snapshot-id"));
+
+        if (snapshot->getValue<Int64>("snapshot-id") == current_snapshot_id)
+        {
+            const auto path = snapshot->getValue<String>("manifest-list");
+            return std::filesystem::path(path).filename();
+        }
+    }
+    return std::nullopt;
+}
+
+
+DataLakeMetadataPtr IcebergMetadata::create(
+    const ObjectStoragePtr & object_storage, const ConfigurationObserverPtr & configuration, const ContextPtr & local_context)
+{
+    auto configuration_ptr = configuration.lock();
+
+    const auto [metadata_version, metadata_file_path] = getMetadataFileAndVersion(object_storage, *configuration_ptr);
+
+    auto log = getLogger("IcebergMetadata");
+    LOG_DEBUG(log, "Parse metadata {}", metadata_file_path);
+
+    StorageObjectStorageSource::ObjectInfo object_info(metadata_file_path);
+    auto buf = StorageObjectStorageSource::createReadBuffer(object_info, object_storage, local_context, log);
+
+    String json_str;
+    readJSONObjectPossiblyInvalid(json_str, *buf);
+
+    Poco::JSON::Parser parser; /// For some reason base/base/JSON.h can not parse this json file
+    Poco::Dynamic::Var json = parser.parse(json_str);
+    const Poco::JSON::Object::Ptr & object = json.extract<Poco::JSON::Object::Ptr>();
+
+    IcebergSchemaProcessor schema_processor;
+
+    auto format_version = object->getValue<int>("format-version");
+
+    auto ptr
+        = std::make_unique<IcebergMetadata>(object_storage, configuration_ptr, local_context, metadata_version, format_version, object);
+
+    return ptr;
+}
+
+ManifestList IcebergMetadata::initializeManifestList(const String & manifest_list_file) const
+{
+    auto configuration_ptr = configuration.lock();
+    if (configuration_ptr == nullptr)
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Configuration is expired");
+
+    auto context = getContext();
+    StorageObjectStorageSource::ObjectInfo object_info(
+        std::filesystem::path(configuration_ptr->getPath()) / "metadata" / manifest_list_file);
+    auto manifest_list_buf = StorageObjectStorageSource::createReadBuffer(object_info, object_storage, context, log);
+
+    LOG_DEBUG(&Poco::Logger::get("initializeManifestList"), "Parse manifest list {}", manifest_list_file);
```

Comment:
Save logger to class? I see it used more than once in the same method, therefore makes sense to save